File: lib/src/main/java/com/llama4j/gguf/PartialModel.java
package com.llama4j.gguf;

import com.llama4j.model.Llama;
import com.llama4j.util.Pair;

import java.util.Map;

record PartialModel(
    String modelFileName,
    Llama model,
    long tensorDataOffset,
    Map<String, GGUFTensorInfo> tensorInfos,
    Pair<float[], float[]> ropeFreqsSWA,
    Pair<float[], float[]> ropeFreqsFull
) {
}

File: lib/src/main/kotlin/com/llama4j/gguf/PartialModel.kt
package com.llama4j.gguf

import com.llama4j.model.Llama

@JvmRecord
internal data class PartialModel(
  val modelFileName: String,
  val model: Llama,
  val tensorDataOffset: Long,
  val tensorInfos: MutableMap<String, GGUFTensorInfo>,
  val ropeFreqsSWA: Pair<FloatArray, FloatArray>,
  val ropeFreqsFull: Pair<FloatArray, FloatArray>
)
